runner/http.go

```diff
@@ -286,6 +286,22 @@ func TransformReader(r io.Reader, fileName string, cti ContentTypeInfo, out io.W
 		}
 
 		return transformORCFile(w.Name(), out)
+	case AvroMimeType:
+		w, err := os.CreateTemp("", "http-avro-temp")
+		if err != nil {
+			return nil
+		}
+		defer os.Remove(w.Name())
+
+		_, err = w.ReadFrom(w)
```

Comment:
There's a typo here. You said `w.ReadFrom(w)` here but I think you meant `w.ReadFrom(r)`.

That said, you don't need to do any of this. The only reason the ORC code wrote to a tempfile was because its library couldn't read from an io.Reader.

You should delete all of this code except for line 304 and change that to

```
return transformAvroFile(r, out)
```

Make sense?

```

Reply:
yeah!